hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java

```diff
@@ -108,7 +108,7 @@ MerkleHederaState newMerkleHederaState(
         return new MerkleHederaState(
                 migration,
                 (event, metadata, provider) -> metadata.app().eventExpansion().expandAllSigs(event, provider),
-                (round, dualState, metadata) -> {
+                (round, state, dualState, metadata) -> {
```

Comment:
With this approach we'll need to add,
```
metaApp.workingStateAccessor().setHederaState(state);
```
...so we always have an up-to-date `WorkingStateAccessor` when we enter handle.

Reply:
Agreed ! Done